detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/BracesOnIfStatements.kt

```diff
@@ -0,0 +1,178 @@
+ * if (a) b else { c; d } // multi-expression
+ * // multiLine = 'consistent'
+ * if (a) { b }
+ * else { c }
+ * if (a) b
+ * else c
+ * </compliant>
+ */
+class BracesOnIfStatements(config: Config = Config.empty) : Rule(config) {
+
+    override val issue = Issue(
+        javaClass.simpleName,
+        Severity.Style,
+        "Braces do not comply with the specified policy",
+        Debt.FIVE_MINS
+    )
+
+    @Configuration("single-line braces policy")
+    private val singleLine: BracePolicy by config(BRACES_NEVER) { BracePolicy.getValue(it) }
+
+    @Configuration("multi-line braces policy")
+    private val multiLine: BracePolicy by config(BRACES_ALWAYS) { BracePolicy.getValue(it) }
+
+    override fun visitIfExpression(expression: KtIfExpression) {
+        super.visitIfExpression(expression)
+        if (expression.parent.parent is KtIfExpression) return
+
+        val branches: List<KtExpression> = walk(expression)
+        validate(branches, policy(expression))
+    }
+
+    private fun walk(expression: KtExpression): List<KtExpression> {
+        val list = mutableListOf<KtExpression>()
+        var current: KtExpression? = expression
+        while (current is KtIfExpression) {
+            current.then?.let { list.add(it) }
+            // Don't add `if` because it's an `else if` which we treat as one unit.
+            current.`else`?.takeIf { it !is KtIfExpression }?.let { list.add(it) }
+            current = current.`else`
+        }
+        return list
+    }
+
+    private fun validate(list: List<KtExpression>, policy: BracePolicy) {
+        when (policy) {
+            BracePolicy.Always -> {
+                if (!list.all { hasBraces(it) }) {
+                    val violator = list.first { !hasBraces(it) }
+                    report(violator, policy)
+                }
+            }
+
+            BracePolicy.Never -> {
+                if (!list.all { !hasBraces(it) }) {
+                    val violator = list.first { hasBraces(it) }
+                    report(violator, policy)
+                }
+            }
+
+            BracePolicy.Consistent -> {
+                val reference = hasBraces(list.first())
+                if (!list.all { hasBraces(it) == reference }) {
+                    val violator = list.first { hasBraces(it) != reference }
+                    report(violator, policy)
+                }
+            }
+        }
+    }
+
+    private fun report(violator: KtExpression, policy: BracePolicy) {
+        val iff = violator.parent.parent as KtIfExpression
+        val reported = when {
+            iff.then === violator -> iff.ifKeyword
+            iff.`else` === violator && isMultiStatement(violator) -> iff.ifKeyword
+            iff.`else` === violator -> iff.elseKeyword
+            else -> error("Violating element ($violator) is not part of this if: $iff")
+        }
+        report(CodeSmell(issue, Entity.from(reported ?: violator), issue.description + ": $policy"))
```

Comment:
Let's be a bit more dev/user-friendly?

```suggestion
        val message = when(policy) {
            Always -> "Missing braces on this branch, add them."
            Never -> "Extra braces exist on this branch, remove them."
            Consistent -> "Inconsistent braces, make sure all branches either have or don't have braces."
        }
        report(CodeSmell(issue, Entity.from(reported ?: violator), message))
```